### DISCUSSION TITLE
[codespace] Retrieve docker images and volumes from prebuilds

### DISCUSSION THREAD

**Discussion by cyppan** [devcontainers/community#24] [Q&A]
Hello! 

I'm trying to setup prebuilds in Codespace and so I'm leveraging the onCreateCommand (and onUpdateContent) lifecycles in the devcontainer  to `docker compose pull` and `docker compose build` (I'm using docker compose inside the devcontainer) to avoid doing so at devcontainer start time, here is my configuration:

devcontainer.json:
```
{
  "build": {
		"dockerfile": "Dockerfile.codespace"
	},
  "runArgs": [
		"--init",
		"--privileged",
		"--cap-add=NET_ADMIN",
		"--cap-add=NET_RAW",
		"--device=/dev/net/tun"
	],
  "overrideCommand": false,
  ...
```

Dockerfile.codespace
```
FROM mcr.microsoft.com/devcontainers/universal:linux
# additional packages installed
...
```

The problem is that the images / volumes are lost in the process when I start my codespace from the prebuild I have to pull and build everything, I understand only the `/workspaces` folder is persisted across builds so I wonder if there is a way to change the docker volume directory of the devcontainer to one in the workspace? I've tried several things but couldn't make it work so far.

Do I miss something or is it an unavoidable limitation? Any other approach possible maybe? Thanks in advance 🙏🏻

**Comment by Chuxel** [comment]
I'd recommend raising the question with the Codespaces folks here: https://github.com/community/community/discussions/categories/codespaces
